From 4dcf1eeaa1742435027ad0cb2a218e7e9b776011 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Mon, 19 Nov 2012 10:19:52 -0200
Subject: [PATCH] qlibraryinfo: Allow the specification of where to load
 qt.conf

This is important to allow cross toolchains to work out of box.

This patch was based on an existent patch in OpenEmbedded.

Change-Id: Idc4feee95a9961db02752d88da9cfe360e89f8e3
---
 src/corelib/global/qlibraryinfo.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
index a756577..0c92e12 100644
--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -164,6 +164,10 @@ QSettings *QLibraryInfoPrivate::findConfiguration()
 	    }
     }
 #endif
+    if (!QFile::exists(qtconfig)) {
+        qtconfig = QFile::decodeName(getenv("QT_CONF_PATH"));
+        printf("Using qt.conf from: %s\n", qtconfig.toLatin1().data());
+    }
     if (QFile::exists(qtconfig))
         return new QSettings(qtconfig, QSettings::IniFormat);
     return 0;     //no luck
-- 
1.7.10.4

